### ISSUE TITLE
How to export private key from keybase with API or kbpgp.js?

### ISSUE THREAD

**Issue by lettergram** [keybase/keybase-issues#1912]
I have the desire to access my private key stored in keybase. However, when I login and view: `user.private_keys`, there doesn't appear that a private key stored the same way as `user.public_keys`.

When I view: `user.private_keys.primary.bundle` there does appear to be a string that could be a private key, but I don't know how to format it into a usable pirvate_key. It looks like the string is either encrypted or encoded differently than normal.

Basically, you can export your public key from keybase, is that functionality available in the API, or some other method I don't know about?

**Comment by oconnor663** [comment]
The string you're seeing is Base64(MessagePack(some stuff + [TripleSec](https://keybase.io/triplesec)(your private key))). You could decrypt it, but it will be inconvenient. You probably want to export your private key with this little menu:

![image](https://cloud.githubusercontent.com/assets/860932/11731264/1225884c-9f68-11e5-8223-496beae27bd5.png)

**Comment by malgorithms** [comment]
If you're running the Go client of Keybase ( keybase.io/download ) this also works:

```bash
keybase pgp export --secret
```

That'll decrypt + export your PGP secret key, which you can pipe into gpg or a file or whatever you want.

**Comment by leonardishere** [comment]
> The string you're seeing is Base64(MessagePack(some stuff + TripleSec(your private key))). You could decrypt it, but it will be inconvenient.

Assuming I have a situation that requires I do it in code, how would I do that? I already have the login code and can retrieve my user object that has the bundled private key, how do I unbundle it?
